src/deploy/functions/prepare.ts

```diff
@@ -53,18 +54,28 @@ export async function prepare(
   options: Options,
   payload: args.Payload
 ): Promise<void> {
-  if (!options.config.src.functions) {
-    return;
+  const projectId = needProjectId(options);
+
+  const sourceDirName = options.config.get("functions.source") as string;
+  if (!sourceDirName) {
+    throw new FirebaseError(
```

Comment:
Be careful here. I think I tried this and we started throwing exceptions for projects that didn't include functions.